hazelcast/src/main/java/com/hazelcast/internal/hotrestart/InternalHotRestartService.java

```diff
@@ -141,4 +155,10 @@ public interface InternalHotRestartService {
      * @param postJoinOp
      */
     void deferPostJoinOps(OnJoinOp postJoinOp);
+
+    void setClusterTopologyStateOnMaster(ClusterTopologyState clusterTopologyState);
+
+    boolean isStartingFromPersistence();
+
+    void onClusterTopologyIntentChange();
```

Comment:
Why do we need to call `onClusterTopologyIntentChange()`? Also can we have Javadocs for the new methods added in this interface.

Reply:
added in 4f840b6f9e7a8b6f775e7ee9951358689ce5d022